onnxruntime/core/providers/shared/utils/utils.cc

```diff
@@ -0,0 +1,70 @@
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "utils.h"
+
+#include <core/common/safeint.h>
+#include <core/graph/graph.h>
+
+#include "core/providers/common.h"
+
+namespace onnxruntime {
+
+NodeAttrHelper::NodeAttrHelper(const onnxruntime::Node& node)
+    : node_attributes_(node.GetAttributes()) {}
+
+float NodeAttrHelper::Get(const std::string& key, float def_val) const {
+  if (!HasAttr(key))
+    return def_val;
+
+  return node_attributes_.at(key).f();
+}
+
+int32_t NodeAttrHelper::Get(const std::string& key, int32_t def_val) const {
+  if (!HasAttr(key))
+    return def_val;
+
+  return SafeInt<int32_t>(node_attributes_.at(key).i());
+}
```

Comment:
Is SafeInt needed here due to a conversion?

Reply:
Added an extra int64_t get() in next PR